internal/services/nginx/nginx_dataplane_apikey_resource_test.go

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package nginx_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/nginx/2024-11-01-preview/nginxapikey"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/nginx"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type DataplaneAPIKeyResource struct{}
+
+func (a DataplaneAPIKeyResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := nginxapikey.ParseApiKeyID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := client.Nginx.NginxApiKey.ApiKeysGet(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving DataplaneAPIKey %s: %+v", id, err)
+	}
+	return utils.Bool(resp.Model != nil), nil
+}
+
+func TestAccDataplaneAPIKeyResource_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, nginx.DataplaneAPIKeyResource{}.ResourceType(), "test")
+	r := DataplaneAPIKeyResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("end_date_time").IsNotEmpty(),
+				check.That(data.ResourceName).Key("hint").IsNotEmpty(),
+			),
+		},
+		data.ImportStep("secret_text"),
+	})
+}
+
+func TestAccDataplaneAPIKeyResource_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, nginx.DataplaneAPIKeyResource{}.ResourceType(), "test")
+	r := DataplaneAPIKeyResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("end_date_time").IsNotEmpty(),
+				check.That(data.ResourceName).Key("hint").IsNotEmpty(),
+			),
+		},
+		data.ImportStep("secret_text"),
+	})
+}
+
+func TestAccDataplaneAPIKeyResource_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, nginx.DataplaneAPIKeyResource{}.ResourceType(), "test")
+	r := DataplaneAPIKeyResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("end_date_time").IsNotEmpty(),
+				check.That(data.ResourceName).Key("hint").IsNotEmpty(),
+			),
+		},
+		data.ImportStep("secret_text"),
```

Comment:
a comment as to why you are ignoring the field will be nice here for future references.